### ISSUE TITLE
Generic Existing Object Mapping

### ISSUE THREAD

**Issue by SunnyVachhetA** [riok/mapperly#1409]
**Is your feature request related to a problem? Please describe.**
I am trying to map existing object from DTO to Model in Entity Framework Core. It works for normal case by declaring void method as described in [Existing Object Mapping](https://mapperly.riok.app/docs/configuration/existing-target/). 

In below code snippet it gives me error of - `RMG001` A mapping method has an unsupported signature.

**Describe the solution you'd like**
API for updating directly working on existing object with Generic Mapping described in https://mapperly.riok.app/docs/configuration/generic-mapping/. 

```csharp
// Relevant new / adjusted API surface
// Expected generated code

[Mapper]
public partial class CarMapper
{
    public partial void Map<TDestination, TDestination>(TSource source, TDestination destination);
    public partial CarDto CarToCarDto(Car car);
    public partial Car CarDtoToCar(CarDto carDto);
}
```

**Describe alternatives you've considered**
I have tried with mentioning return type. But it gives error of method must have an implementation.

**Comment by latonz** [comment]
How would you implement this method manually?

**Comment by SunnyVachhetA** [comment]
> How would you implement this method manually?

For non-generic, It uses the existing object for mapping without creating new one.

Following that I can think of using the same instance of `TDestination` passed for mapping source entity.

TSource may or may not have property that are available in TDestination. Some property values available in TDestination should be override null by TSource.

I am thinking by above approach. Please note that I am learning about source generator, and I may have unexplored areas in this section.

**Comment by SunnyVachhetA** [comment]
Yes @latonz, I was not referring to queryable projection. Generic existing object mapping for class or records only.

Without source generator, I will create a method that takes a existing destination object reference and second one DTO or other object, map them and return the same destination object reference.

**Comment by ste4net** [comment]
This was exactly what i was searching for... i reach this page from the RMG001 error and with this post i understand it it not possible to do what i was trying to do.
Just to add additional information i was searching for something like this.

```
public static partial void Map<Tsrc, Ttar>(Tsrc source, Ttar destination);
public static partial void CarToCarDto(Car src, CarDto target);
public static partial void CarDtoToCar(CarDto src, Car target);
```

with generated code..

 
 ```
    public static void Map<Tsrc, Ttar>(Tsrc source, Ttar targ)
    {
        Type targetType = typeof(Ttar);
        switch (source)
        {
            case Car c when targetType.IsAssignableFrom(typeof(CarDto)):
                CarToCarDto(source as Car, targ as CarDto);
                break;
            case CarDto c when targetType.IsAssignableFrom(typeof(Car)):
                CarDtoToCar(source as CarDto,targ as Car);
                break;
        };
    }
```
